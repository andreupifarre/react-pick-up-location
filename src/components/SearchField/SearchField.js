import React, { useState } from 'react';
import './SearchField.scss';

const SearchField = () => {
  const [searchText, setSearchText] = useState('');

  return (
    <div className="c-search-field">
      <label htmlFor="c-search-input" className="c-search-field__label">
        Pick-up Location
      </label>
      <input
        id="c-search-input"
        type="text"
        className="c-search-field__input"
        autoComplete="off"
        placeholder="city, airport, station, region, districtâ€¦"
        aria-required="true"
        aria-autocomplete="list"
        aria-haspopup="true"
        onChange={e => setSearchText(e.target.value)}
      />
    </div>
  );
};

export default SearchField;
