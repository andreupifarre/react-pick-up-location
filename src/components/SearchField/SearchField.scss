.c-search-field {
  position: relative;

  &__label {
    display: inline-block;
    padding: 0 0 6px;
    color: #444;
    font-size: 14px;
  }

  &__input {
    -webkit-appearance: none;
    border: 1px solid #a4a4a4;
    border-radius: 4px;
    min-width: 49px;
    padding: 8px 16px;
    width: 100%;
    height: 48px;
    font-size: 1rem;
    border-color: #c8ae5b;
    font-family: Ubuntu, Open Sans, sans-serif;
    font-weight: normal;

    &:focus {
      border-color: #0f6193;
      outline: 0;
    }

    &::placeholder {
      font-size: 14px;
    }
  }
}
