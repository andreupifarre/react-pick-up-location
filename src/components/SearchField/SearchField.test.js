import React from 'react';
import { shallow } from 'enzyme';

import SearchField from './SearchField';

const render = () => shallow(<SearchField />);

describe('SearchField', () => {
  it('should contain a parent element with a class .c-search-field', () => {
    const element = render().find('.c-search-field');
    expect(element.exists()).toBeTruthy();
  });

  it('should contain a label element with a class .c-search-field__label', () => {
    const element = render().find('.c-search-field__label');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('label');
  });

  it('should contain an input field with a class .c-search-field__input', () => {
    const element = render().find('.c-search-field__input');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('input');
  });

  it('should have accessibility and other required attributes', () => {
    const element = render().find('.c-search-field__input');
    expect(element.props()).toEqual({
      ...element.props(),
      onBlur: expect.any(Function),
      onFocus: expect.any(Function),
      onChange: expect.any(Function),
      placeholder: 'city, airport, station, region, districtâ€¦',
      autoComplete: 'off',
      'aria-required': 'true',
      'aria-autocomplete': 'list',
      'aria-haspopup': 'true',
    });
  });

  it('should contain a SearchList component', () => {
    const wrapper = render();
    wrapper.find('input').simulate('focus');

    const element = wrapper.find('SearchList');
    expect(element.exists()).toBeTruthy();
    expect(element.prop('locations')).toBeDefined();
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
