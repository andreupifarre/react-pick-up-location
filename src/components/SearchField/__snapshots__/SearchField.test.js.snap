// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField should render correctly 1`] = `
Array [
  <div
    className="c-search-field"
  >
    <label
      className="c-search-field__label"
      htmlFor="c-search-input"
    >
      Pick-up Location
    </label>
    <input
      aria-autocomplete="list"
      aria-haspopup="true"
      aria-required="true"
      autoComplete="off"
      className="c-search-field__input"
      id="c-search-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="city, airport, station, region, districtâ€¦"
      type="text"
    />
  </div>,
]
`;
